ivy/functional/backends/jax/general.py

```diff
@@ -67,11 +67,11 @@ def to_list(x: JaxArray) -> list:
     return _to_array(x).tolist()
 
 
-def shape(x: JaxArray, as_array: bool = False) -> Union[tuple, JaxArray]:
+def shape(x: JaxArray, as_array: bool = False) -> Union[ivy.Shape, ivy.Array]:
     if as_array:
-        return jnp.asarray(jnp.shape(x))
+        return ivy.array(x.shape)
```

Comment:
actually we should have `ivy.array(jnp.shape(x))` or `ivy.Array(jnp.shape(x))` if `jnp.shape` returns a native shape instance.